oonf-olsrd2: update to current develop branch - fixes routing#899

XDjackieXD opened the pull request:
Development of OONF is very slow. The master branch is completely out of date and new releases haven't been tagged since 2018. According to this PR https://github.com/OLSR/OONF/pull/49#issuecomment-1229195523 the authors will probably just abandon the current master branch and replace it by the develop branch. During my testing, the current development branch is more stable than the latest tagged version on current OpenWrt versions.

Maintainer: me
Compile tested: Ubiquiti ER-X-SFP, MT7621 with both OpenWrt 22.03 and latest master.
Run tested: Ubiquiti ER-X-SFP, MT7621 with both OpenWrt 22.03 and latest master. 22.03 running stable since multiple weeks on my roof.

BKPepe commented on oonf-olsrd2/Makefile:
```diff
@@ -1,13 +1,15 @@
 PKG_NAME:=oonf-olsrd2
-PKG_RELEASE:=1
+PKG_VERSION:=v0.15.99
+PKG_REV:=1cec9b21086fb52ab4262c69aabd087e4d2d3a44
```
This variable is deprecated. No need to use it anymore.

BKPepe commented on oonf-olsrd2/Makefile:
```diff
@@ -36,13 +38,14 @@ CMAKE_OPTIONS+=-D CMAKE_BUILD_TYPE:String=$(BUILD_TYPE) \
                -D INSTALL_INCLUDE_DIR:Path=include/oonf \
                -D INSTALL_CMAKE_DIR:Path=lib/oonf \
                -D CMAKE_PREFIX_PATH=$(STAGING_DIR)/usr
+#              -D CMAKE_GENERATOR=Ninja
```
Is there any reason, why it is not possible to use Ninja?

XDjackieXD replied:
It wasn't active before. It probably works but I did not want to activate it when it wasn't active before ^^

XDjackieXD replied:
Ninja builds seem to work fine, I'll enable Ninja.

BKPepe commented on oonf-olsrd2/Makefile:
```diff
@@ -1,13 +1,15 @@
-PKG_RELEASE:=1
+PKG_VERSION:=v0.15.99
+PKG_REV:=1cec9b21086fb52ab4262c69aabd087e4d2d3a44
+PKG_RELEASE:=$(PKG_REV)
```
```suggestion
PKG_RELEASE:=1
```

XDjackieXD replied:
Is there an easy way to get the short commit ref into the package version/release?
As this isn't a release version it will probably update in the future without a change in the version number and I think having the shortref instead of a in incrementing number (that we'll need to manually update) as the release would make more sense (kinda like it is for the olsr1 package but just entering "$(AUTORELEASE)" here did not seem to work)

BKPepe replied:
Sure, just drop ``PKG_VERSION`` and that's it. :)

XDjackieXD commented:
Updated the Makefile with your suggestions and added a patch to fix the lan_import plugin build (and I'll prepare a pr for upstream to fix this there...)

BKPepe commented:
> (and I'll prepare a pr for upstream to fix this there...)

Please do and once it is done, I will merge this here.

XDjackieXD commented:
https://github.com/OLSR/OONF/pull/62 PR on OONF is now open to fix the lan_import build on the develop branch

BKPepe commented:
CI fails:
```
WARNING: PKG_MIRROR_HASH does not match oonf-olsrd2-2022-08-25-1cec9b21.tar.xz hash e9065af0a7103e84b9084683bb1db87c3361c1d97d8ad8a48a021b904b35ed56
```

XDjackieXD commented:
Ah sorry yes I still had the hash of my testing version in there. weirdly enough my local builds don't complain about the mismatch o.O

BKPepe commented:
You need to refresh the patch now, that's what can be done by ``make package/oonf-olsrd2/refresh V=s``.

BKPepe commented:
Also, please assign your email address, which you used in the commit to your GitHub account.
